crates/red_knot_python_semantic/src/types/display.rs

```diff
@@ -90,6 +90,24 @@ impl Display for DisplayRepresentation<'_> {
 
                 escape.bytes_repr().write(f)
             }
+            Type::SliceLiteral(slice) => {
+                write!(
+                    f,
+                    "slice[{start}, {stop}{step}]",
+                    start = slice
+                        .start(self.db)
+                        .map(|s| format!("Literal[{s}]"))
+                        .unwrap_or("None".into()),
+                    stop = slice
+                        .stop(self.db)
+                        .map(|s| format!("Literal[{s}]"))
+                        .unwrap_or("None".into()),
+                    step = slice
+                        .step(self.db)
+                        .map(|s| format!(", Literal[{s}]"))
+                        .unwrap_or_default()
+                )
```

Comment:
This code isn't hot so unlikely that it's worth it but it's unfortunate that we have to allocate a `String` just to write the literal
```suggestion
							f.push_str("slice[")?;
							if let Some(start) = slice.start(self.db) {
								write!(f, "Literal[{start}]")?;
							} else {
								f.push_str("None")?;
							}
							
							f.push_str(", ")?;
							
							if let Some(stop) = slice.stop(self.db) {
								write!(f, "Literal[{stop}]")?;
							} else {
								f.push_str("None")?;
							}
							
							if let Some(step) = slice.step(self.db) {
								write!(f, ", Literal[{step}])?;
							}
```